### PR TITLE
Run `pydocstyle` in pre-commit

### LINKED ISSUE #6378: Add pydocstyle hook to pre-commit
Mentioned in https://github.com/pymc-devs/pymc/pull/6374#discussion_r1043097788_

### PR DESCRIPTION
**Pull request by michaelosthege**
This runs `pydocstyle` in the pre-commit and fixes a bunch of things it complained about.

However, I had to silence many errors that were just too common:

| id | occurrences | description |
|-|-|-|
| D100 | 62 | Missing docstring in public module |
| D101 | 62 | Missing docstring in public class |
| D102 | 568 | Missing docstring in public method |
| D103 | 98 | Missing docstring in public function |
| D104 | 10 | Missing docstring in public package |
| D105 | 47 | Missing docstring in magic method |
| D107 | 84 | Missing docstring in __init__ |
| D200 | 23 | One-line docstring should fit on one line with quotes |
| D202 | 59 | No blank lines allowed after function docstring |
| D203 | 254 | 1 blank line required before class docstring |
| D204 | 47 | 1 blank line required after class docstring |
| D205 | 155 | 1 blank line required between summary line and description |
| D209 | 16 | Multi-line docstring closing quotes should be on a separate line |
| D212 | 231 | Multi-line docstring summary should start at the first line |
| D213 | 330 | Multi-line docstring summary should start at the second line |
| D301 | 30 | Use r""" if any backslashes in a docstring |
| D400 | 349 | First line should end with a period |
| D401 | 98 | First line should be in imperative mood |
| D403 | 14 | First word of the first line should be properly capitalized |
| D413 | 341 | Missing blank line after last section |
| D415 | 334 | First line should end with a period, question mark, or exclamation point |
| D417 | 089 | Missing argument descriptions in the docstring |

Closes #6378

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- None

## Bugfixes / New features
- None

## Docs / Maintenance
- Now running `pydocstyle` in the pre-commit
- Fixed several docstring formatting issues.

### CODE REVIEW

**Comment by ricardoV94** [comment]
Except for the D100s... it seems like we should have the others enabled and they shouldn't be too hard to fix no?

**Review by OriolAbril — DISMISSED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -35,6 +35,14 @@ repos:
       args: [--rcfile=.pylintrc]
       files: ^pymc/
       exclude: (?x)(pymc/_version.py)
+- repo: https://github.com/PyCQA/pydocstyle
+  rev: 6.1.1
+  hooks:
+    - id: pydocstyle
+      args:
+      - --ignore=D100,D101,D102,D103,D104,D105,D107,D200,D202,D203,D204,D205,D209,D212,D213,D301,D400,D401,D403,D413,D415,D417
```
Comment (OriolAbril) [inline .pre-commit-config.yaml:43]: The convention arg is missing, we should be using `numpy` as convention (which should remove the contradictory warnings happening all at the same time)

Comment (OriolAbril) [reply .pre-commit-config.yaml:43]: e.g. D212 and D213 make no sense together.

**Comment by OriolAbril** [comment]
I have also just realised we have https://github.com/pymc-devs/pymc/blob/main/setup.cfg#L7 which seems to be ignored for some reason? maybe we ought to install `pydocstyle[toml]`?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (michaelosthege) [reply .pre-commit-config.yaml:43]: Just tried it, but with `--convention` one can't pass `--ignore` and then we'd have hundreds of errors and would need to fix them all in one go (unrealistic..)

Comment (OriolAbril) [reply .pre-commit-config.yaml:43]: let's keep ignore then

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**